### ISSUE TITLE
有没有ParametricLIFNode的Demo?

### ISSUE THREAD

**Issue by ketechn** [fangwei123456/spikingjelly#360]
另外没太看明白，tau是如何训练的，可否指点，谢谢

**Comment by fangwei123456** [comment]
自动训练的参数是pytorch的机制，建议学习一下`torch.nn.Parameter`的原理。

**Comment by ketechn** [comment]
好的，另外，ParametricLIFNode类中为什么没有single_step_forward,jit_eval_single_step_forward,jit_eval_multi_step_forward这些实现？谢谢！

**Comment by ketechn** [comment]
> 自动训练的参数是pytorch的机制，建议学习一下`torch.nn.Parameter`的原理。

好的，另外，ParametricLIFNode类中为什么没有single_step_forward,jit_eval_single_step_forward,jit_eval_multi_step_forward这些实现？谢谢！

**Comment by fangwei123456** [comment]
jit函数是为了加速做的，LIF和IF是最常用的，因此只给这2个单独写了

**Comment by AllenYolk** [comment]
`ParametricLIFNode` 中可学习参数 `w` 的训练由 PyTorch 的 autograd 自动处理——只要 `w` 被声明为 `nn.Parameter`，PyTorch 会自动计算它对 loss 的梯度并由 optimizer 更新。

`ParametricLIFNode` 内部把 `w` 包装为 `nn.Parameter`，并定义：

```python
self.w = nn.Parameter(torch.tensor(init_w, dtype=torch.float))
```

其中 `init_w = math.log(init_tau / (init_tau - 1.0))` 是为了让 `1/tau = Sigmoid(w)` 在 `init_tau` 附近取到正确初值。

用法和普通 `nn.Module` 的可学习参数一样：

```python
node = neuron.ParametricLIFNode(init_tau=2.0)
optimizer = torch.optim.Adam(node.parameters(), lr=1e-3)
```

至于 `single_step_*` JIT 函数：只为 LIF / IF 这两个最常用的基础神经元单独写了，`ParametricLIFNode` 没有单独的 JIT 路径，走的是通用 multi-step kernel。如果对速度敏感，可以在多步模式下使用 `ParametricLIFNode` + `backend="triton"` 或 `"cupy"`。
